interface/ceed-basis.c

```diff
@@ -188,35 +187,15 @@ int CeedBasisCreateH1(Ceed ceed, CeedElemTopology topo, CeedInt ncomp,
                       const CeedScalar *grad, const CeedScalar *qref,
                       const CeedScalar *qweight, CeedBasis *basis) {
   int ierr;
-  CeedInt P = ndof, Q = nqpts, dim = 1;
+  CeedInt P = ndof, Q = nqpts, dim = 0;
 
   if (!ceed->BasisCreateH1)
     return CeedError(ceed, 1, "Backend does not support BasisCreateH1");
   ierr = CeedCalloc(1,basis); CeedChk(ierr);
 
-  switch (topo) {
-  case CEED_LINE:
-    dim = 1;
-    break;
-  case CEED_QUAD:
-    dim = 2;
-    break;
-  case CEED_HEX:
-    dim = 3;
-    break;
-  case CEED_TRIANGLE:
-    dim = 2;
-    break;
-  case CEED_TET:
-    dim = 3;
-    break;
-  case CEED_PYRAMID:
-    dim = 3;
-    break;
-  case CEED_PRISM:
-    dim = 3;
-    break;
-  }
+  for (int i=1; i<=3; i++)
+    if (topo & CeedIntPow(2, i + 2))
+      dim = i;
```

Comment:
We should have a function, even if internal, that gives the dimension for any topology.

Reply:
The way I have it right now, I am imagining that 12-15 are user 1D topologies, 20-23 are user 2D topologies, and 36-39 are user 3D topologies. That way we can support up to 4 additional topologies of each dimension and then I can make an internal function to replicate those three lines.The only part I'm not clear on is how  best to assign these user topologies to these values.

Reply:
But why have a small number of user-defined topologies when it's no harder (arguably easier) to have thousands as my previous suggestion of `dim << 16 | counter` would do?

Reply:
Because I didn't understand how that expression worked and took a stab at making something similar?

Reply:
Ah, `dim << 16` is left-shift by 16 bits. The `|` is bitwise or, which in this case acts like addition. Printed in hexadecimal, `3 << 16 | 5 == 0x30005`. You get the dimension back with `topo >> 16`. Use an unsigned type.